pkg/cloudprovider/providers/openstack/metadata.go

```diff
@@ -85,6 +85,8 @@ func getMetadataFromConfigDrive() (*Metadata, error) {
 			return nil, err
 		}
 		dev = strings.TrimSpace(string(out))
+	} else if err != nil {
```

Comment:
I don't think we want this additional case.

This case triggers if there is an error calling `stat(/dev/disk/by-label/config-2)` but the error is something _other than_ not-found.
For every other error (eg: "permission denied"), it's perfectly safe (and even reasonable) to continue on and try to mount `/dev/disk/by-label/config-2` - and then fail iff *that* operation throws an error.